apps/typegpu-docs/src/content/examples/simple/vaporwave/index.ts

```diff
@@ -0,0 +1,249 @@
+import tgpu from 'typegpu';
+import * as d from 'typegpu/data';
+import * as std from 'typegpu/std';
+import { sdPlane } from '@typegpu/sdf';
+import { perlin3d } from '@typegpu/noise';
+
+import { circles, grid } from './floor.ts';
+import * as c from './constants.ts';
+import { Ray } from './types.ts';
+import { getSphere } from './sphere.ts';
+import { rayUnion } from './helpers.ts';
+
+// == INIT ==
+const canvas = document.querySelector('canvas') as HTMLCanvasElement;
+const context = canvas.getContext('webgpu') as GPUCanvasContext;
+const presentationFormat = navigator.gpu.getPreferredCanvasFormat();
+
+const root = await tgpu.init();
+
+context.configure({
+  device: root.device,
+  format: presentationFormat,
+  alphaMode: 'premultiplied',
+});
+
+// == BUFFERS ==
+const floorAngleBuffer = root.createUniform(d.f32);
+const sphereAngleBuffer = root.createUniform(d.f32);
+const glowIntensityBuffer = root.createUniform(d.f32);
+const resolution = root.createUniform(d.vec2f);
+
+let floorSpeed = 1;
+let sphereSpeed = 1;
+let glowIntensity = 0.14;
+
+let sphereColor = d.vec3f(0, 0.25, 1);
+const sphereColorBuffer = root.createUniform(d.vec3f);
+sphereColorBuffer.write(sphereColor);
```

Comment:
Since we're not using the sphere color on the CPU, just updating the buffer, we don't have to keep it in a variable
Also, tiny nit, the convention is to suffix uniforms with `Uniform`
```suggestion
const sphereColorUniform = root.createUniform(d.vec3f, d.vec3f(0, 0.25, 1));
```